pytorch_lightning/trainer/trainer.py

```diff
@@ -382,10 +382,20 @@ def __init__(
         self.slurm_connector = SLURMConnector(self)
         self.tuner = Tuner(self)
 
-        # max_epochs won't default to 1000 if max_steps/max_time are specified (including being set to -1).
+        if max_steps is None:
+            rank_zero_deprecation(
+                "Setting `max_steps = None` is deprecated in v1.5 and will be removed in v1.7."
+                " Use `max_steps = -1` instead."
+            )
+            max_steps = -1
+
+        if max_epochs is None:
```

Comment:
same here. The Trainer init should be the cleanest part of the codebase ;)

Reply:
This is needed in order to initialize the `FitLoop` (it isn't error checking). Is it okay to leave it?